pywt/_thresholding.py

```diff
@@ -123,3 +163,82 @@ def threshold(data, value, mode='soft', substitute=0):
                 sorted(thresholding_options.keys()))
         raise ValueError("The mode parameter only takes values from: {0}."
                          .format(', '.join(keys)))
```

Comment:
would be good to add a `garotte` example, all the other modes have one (http://pywavelets.readthedocs.io/en/latest/ref/thresholding-functions.html)